docs/frontend/preact.md

```diff
@@ -12,3 +12,72 @@ Preact code is located in `app/javascript`.
 
 The components are mounted when needed, look for `javascript_pack_tag` in the
 view pages inside `app/views`.
+
+## PropTypes
+
+Preact supports
+[PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html). When
+creating Preact components, ensure to always have prop types defined.
+
+### Common PropTypes
+
+PropTypes can become repetitive. In simple cases where a PropType is a string or
+number, this duplication is expected. For commonly used entities in the DEV
+project, common PropTypes exist. All common PropTypes are located in the
+`/app/javascript/src/components/common-prop-types` folder. For example, the user
+entity that is referenced in many parts of the DEV codebase.
+
+```javascript
+import PropTypes from 'prop-types';
+
+export const userPropTypes = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  profile_image_url: PropTypes.string.isRequired,
+  summary: PropTypes.string.isRequired,
+});
+```
+
+#### Using Common PropTypes
+
+Common PropTypes are imported just like any other
+[JavaScript Module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules).
+For example, here are two scenarios where a component needs to use the
+`tagPropTypes`.
+
+Example 1: Using the `tagPropTypes` with `PropTypes.arrayOf` to construct the
+PropTypes for the `tags` prop.
```

Comment:
It might be nice to explain why someone might need to do this/when you'd want to do something like this.

Reply:
I'll add a little blurb to explain. Thanks for all the great feedback! 🔥